pages/learn/develop/container-contracts.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Container Contracts
+excerpt: A guide to using container contracts to enforce device compatibility
+---
+
+# Container contracts
+
+__Note__ Container contracts are available for devices running Supervisor versions >= 10.6.0. All prior versions will not enforce any contracts.
+
+Container contracts are used to ensure the compatibility of a device to run an application service. For example, container contracts may be used to ensure that the host operating system is running a specific version of balenaOS or Supervisor such that if an application is dependent on a specific version, it only gets deployed to those devices that can meet these requirements to fulfill the contract. Currently, you may define contract requirements based on the versions of [balenaOS](/reference/OS/overview/2.x/) and [Supervisor](/reference/supervisor/supervisor-api/).
+
+Each service container can define the contracts that it enforces. If a container's contract requirements are not met, the release is not deployed to the device. You only need to define a contract for service containers that you wish to enforce a contract.
+
+## Contract specification
+
+Container contracts are defined as a `contract.yml` file and must be placed in the application structure alongside the corresponding service Dockerfile. For example, the following [multi-container application](/learn/develop/multicontainer/) defines two services named `first-service` and `second-service`, each with a corresponding `contract.yml` file.
+
+```shell
+.
+├── docker-compose.yml
+├── first-service
+│   ├── contract.yml
+│   └── Dockerfile.template
+└── second-service
+    ├── contract.yml
+    └── Dockerfile.template
+```    
+
+`contract.yml` is a YAML file containing `type`, `slug`, `name`, and `requires` keys, which define the requirements of the contract. The following example requires the device to be running a version of balenaOS greater or equal than `2.40.0` for the contract to pass.
+
+```yaml
+type: 'sw.container'
+slug: 'enforce-os-gt-240'
+name: 'Enforce hostOS Version'
+requires:
+  - type: 'sw.os'
+    version: '>=2.40.0'
+```
+
+* `type` should be set to `sw.container` for container contracts.
+* `slug` should be a unique identifier to identify the contract.
+* `name` should be a descriptive name of the contract.
+* `requires` should be a list of requirements to be enforced by the contract defined by `type` and `version` keys. Currently, `type` may be either `sw.os` for the hostOS version or `sw.supervisor` for the device Supervisor version. 
+
+__Note__ If `name` or `type` are not supplied, the build will fail. If `slug` is not provided, the build will complete, but the contract will fail validation on deployment. If `requires` is not provided, the contract will have no effect and the release will be deployed.
```

Comment:
I'm going to fix this so maybe change it to "if `name`, `type` or `slug` are ommitted, the build will fail".

Reply:
Perfect, yeh I added here while I noted it but was a little prompt for this to be improved 👍